thesis/inc/introduction/game_industry.tex

```diff
@@ -0,0 +1,34 @@
+\section{Game Industry}
+As developers of quality interactive experiences, modern video game developers face a fairly unique
+combination of challenges.
+
+\subsection{Flexibility}
+A valued concept within the video game industry is fast iteration cycles.
+Game developers often want to be able to test out and iterate on their ideas quickly.
+Fast iteration cycles require a great deal of flexibility,
+it is therefore desirable to keep software modules decoupled and generic,
+allowing developers to quickly combine different modules into new combinations creating new functionality,
+and content.
+
+\subsection{Re-usability}
+Similar; but not equal to flexibility, are the desire for re-usability.
+Like many other branches of software, the game industry are interested in reusable and flexible
+software modules, to save both money and development time.
+
+\subsection{Performance}
+Jason Gregory\cite[p. 9]{game_engine_architecture} describes video games as "soft real-time interactive agent-based computer simulations".
+As with all real-time systems certain deadlines need to be met, the most obvious example being frame rate requirements.
+Video games are also expected to push limits, constantly improving areas like graphics, physics simulations, AI, etc.
+Potentially for multiple platforms with different strengths and weaknesses.
+
+These desires are often at odds with each other, and finding a balance is a difficult task.
+Additional factors also affect the balancing of these desires, like the type of game that is being developed,
+or what platforms the game is being developed for.
+
+\subsection{Game Engines}
+Modern video games are often created using game engines. A game engine can be seen as a foundation/framework
+containing a set of core modules needed for a game, like a rendering engine, or physics engine.
+Different game engines have different strengths and weaknesses, depending on different factors,
+examples include what sort of game genre the engine was designed for.
```

Comment:
could also talk about engines for different platforms, and for different level of programming competence. Would say that unity is much more beginner friendly than unreal.